### PR TITLE
Checking for `@type` being an array

### LINKED ISSUE #598: Checking for TM is too restrictive
As seen at https://github.com/eclipse-thingweb/playground/blob/f07db5a1e6474deadf2506dd56bcadf2a56e0e99/packages/web/src/scripts/editor.js#L291 , the code checks if `@type` is equal to `tm:ThingModel`. However, it should check whether `tm:ThingModel` is in the array or not.

### PR DESCRIPTION
**Pull request by egekorkan**
fixes #598

### CODE REVIEW

**Comment by egekorkan** [comment]
Gotten a bit rusty and made some mistakes... Will refactor :)

File: packages/web/src/scripts/util.js [hunk 1]
```diff
@@ -580,17 +607,44 @@ function getEndPositionOfMatch(match) {
...
+    //TODO: Move to core package after refactoring
+    if ( "@type" in jsonDocument ){
+        if (typeof jsonDocument["@type"] === "string") {
+            if (jsonDocument["@type"] === "tm:ThingModel") {
+                return "tm";
+            } else {
+                return "td";
+            }
+        } else {
+            // then it is an array
```
Comment (danielpeintner) [inline packages/web/src/scripts/util.js:640]: Wouldn't it be better to check for array?

File: packages/web/src/scripts/util.js [hunk 2]
```diff
@@ -580,17 +607,44 @@ function getEndPositionOfMatch(match) {
...
     }
 
-    return monacoLocation
+    return monacoLocation;
 }
+
+/**
+ * Checks if the input is a TD or TM
+ * @param {jsonDocument} 
+ * @returns "tm" or "td"
```
Comment (danielpeintner) [inline packages/web/src/scripts/util.js:628]: Would it make sense to report a unknown value also so the caller can decide what it is defaulting to?

**Review by danielpeintner — COMMENTED** [review]
Just some comments

**Pull request merged**